src/matchers/tables.lua

```diff
@@ -1,13 +1,122 @@
+local prettyValue = require("src.utils.prettyValue")
+local deepEqual = require("src.utils.deepEqual")
+local assertType = require("src.asserts.type")
+
 ---@type lest.Matcher
-local function toHaveLength(ctx, received, length) end
+local function toHaveLength(ctx, received, length)
+	assertType(length, "number")
+
+	local success, comparison = pcall(function()
+		return #received == length
+	end)
+
+	return {
+		pass = success and comparison,
+		message = string.format(
+			"Expected %s to%shave a length of %d",
+			prettyValue(received),
+			ctx.inverted and " not " or " ",
+			length
+		),
+	}
+end
 
 ---@type lest.Matcher
-local function toContain(ctx, received, item) end
+local function toContain(ctx, received, item)
+	local pass = false
+
+	if type(received) == "string" then
+		pass = received:find(tostring(item), 1, true) ~= nil
```

Comment:
It doesn't make sense to allow any type to be contained within a string, we should simply set `pass` to `false` if `received` is a string and `item` isn't

Reply:
Fuck you